apps/root/src/pages/aggregator/swap-container/index.tsx

```diff
@@ -69,32 +79,28 @@ const SwapContainer = () => {
   }, [mappedNetworks]);
 
   React.useEffect(() => {
-    if (!from) {
-      if (fromParamToken) {
-        dispatch(setFrom(fromParamToken));
-      } else if (fromParamCustomToken && !from) {
-        dispatch(setFrom(fromParamCustomToken.token));
-      } else if (!from && !to && !toParamToken && !toParamCustomToken) {
-        let networkToUse = find(mappedNetworks, { chainId: Number(chainId) });
-        if (!networkToUse && chainId) {
-          networkToUse = find(mappedNetworks, { name: chainId.toLowerCase() });
-        }
-        dispatch(
-          setFrom(
-            getProtocolToken(Number(networkToUse?.chainId || actualCurrentNetwork.chainId || currentNetwork.chainId))
-          )
-        );
-      }
+    const handleCustomTokens = (tokenAddress: Address) => {
+      void handleCustomTokenBalances({ tokenAddress, chainId: networkToUse });
+    };
+    if (!isLoadingAllTokenLists && !fromParamToken && fromParam && isAddress(fromParam)) {
+      handleCustomTokens(fromParam);
+    }
+    if (!isLoadingAllTokenLists && !toParamToken && toParam && isAddress(toParam)) {
+      handleCustomTokens(toParam);
+    }
+  }, [isLoadingAllTokenLists]);
+
+  React.useEffect(() => {
+    if ((!from && fromParamToken) || (fromParamToken && prevIsLoadingCustomToken && !isLoadingCustomToken)) {
+      dispatch(setFrom(fromParamToken));
```

Comment:
We might have some unwanted re-renders here if the custom token search is modified since its going to modify this on the top and the isLoading params are going to change, thus entering here again an dispatching

Reply:
I have re-defined the customToken hook so we can instance it once per token. This useEffect should not be triggered if the search changes, it's only meant to handle the url params. So the if blocks are only executed when a custom token from url param was loading in a previous render, but now it's not loading anymore.

Also, the isLoading for custom tokens in this component should only be triggered once, when the tokenList has loaded